tests/functional/list/test_list.py

```diff
@@ -0,0 +1,579 @@
+import pytest
+import os
+import json
+
+from dbt.tests.util import run_dbt
+from dbt.logger import log_manager
+
+from tests.functional.list.fixtures import (  # noqa: F401
+    snapshots,
+    tests,
+    models,
+    macros,
+    seeds,
+    analyses,
+    project_files,
+)
+
+
+class TestList:
+    def dir(self, value):
+        return os.path.normpath(value)
+
+    @pytest.fixture(scope="class")
+    def model_path(self):
+        return self.dir("models")
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "analysis-paths": [self.dir("analyses")],
+            "snapshot-paths": [self.dir("snapshots")],
+            "macro-paths": [self.dir("macros")],
+            "seed-paths": [self.dir("seeds")],
+            "test-paths": [self.dir("tests")],
+            "seeds": {
+                "quote_columns": False,
+            },
+        }
+
+    def run_dbt_ls(self, args=None, expect_pass=True):
+        log_manager.stdout_console()
```

Comment:
I think we have a function called `run_dbt_and_capture` where `run_dbt` is defined that does similar thing, but this also works

Reply:
I played around with `run_dbt_and_capture` a little bit, however tests failed and I couldn't immediately find the reason, so I'm going to leave this as it for now.